<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="base_frame" default="base_link"/>
  <arg name="lost_frame_max_acceleration" default="10.0"/>
  <arg name="calibration_skip_keyframes" default="7"/>
  <arg name="mapping_registrator" default="ndt" description="ndt or gicp"/>
  <arg name="rviz" default="true"/>

  <let name="calibration_camera_optical_link_frames" value="['']"/>

  <let name="calibration_lidar_frames" value="[
    velodyne_left,
    velodyne_right]"/>

  <let name="mapping_lidar_frame" value="velodyne_top"/>
  <let name="mapping_pointcloud" value="/sensing/lidar/top/pointcloud"/>
  <let name="detected_objects" value="/perception/object_recognition/detection/objects"/>

  <let name="calibration_camera_info_topics" value="['']"/>

  <let name="calibration_image_topics" value="[
    '']"/>

  <let name="calibration_pointcloud_topics" value="[
    /sensing/lidar/left/rectified/pointcloud_ex,
    /sensing/lidar/right/rectified/pointcloud_ex]"/>

  <!-- mapping based calibrator -->
  <include file="$(find-pkg-share extrinsic_mapping_based_calibrator)/launch/calibrator.launch.xml">
    <arg name="ns" value=""/>
    <arg name="calibration_service_name" value="calibrate_lidar_lidar"/>

    <arg name="rviz" value="$(var rviz)"/>
    <arg name="base_frame" value="$(var base_frame)"/>

    <arg name="calibration_camera_optical_link_frames" value="$(var calibration_camera_optical_link_frames)"/>
    <arg name="calibration_lidar_frames" value="$(var calibration_lidar_frames)"/>
    <arg name="mapping_lidar_frame" value="$(var mapping_lidar_frame)"/>

    <arg name="mapping_pointcloud" value="$(var mapping_pointcloud)"/>
    <arg name="detected_objects" value="$(var detected_objects)"/>

    <arg name="calibration_camera_info_topics" value="$(var calibration_camera_info_topics)"/>
    <arg name="calibration_image_topics" value="$(var calibration_image_topics)"/>
    <arg name="calibration_pointcloud_topics" value="$(var calibration_pointcloud_topics)"/>

    <arg name="mapping_registrator" value="$(var mapping_registrator)"/>
    <arg name="lost_frame_max_acceleration" value="$(var lost_frame_max_acceleration)"/>
    <arg name="calibration_skip_keyframes" value="$(var calibration_skip_keyframes)"/>
  </include>
</launch>
